tests/integration/ember-engines-test.js

```diff
@@ -0,0 +1,50 @@
+import { module, test } from 'qunit';
+import {
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+  teardownRenderingContext,
+  render,
+} from '@ember/test-helpers';
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+import hbs from 'htmlbars-inline-precompile';
+
+const modulePrefix = 'eager-engine';
+const resolver = engineResolverFor(modulePrefix);
+
+module('setupRenderingContext for "ember-engines"', function (hooks) {
+  hooks.beforeEach(async function () {
+    await setupContext(this, { resolver });
+    // this.owner now is an engine.
+    this.engine = this.owner;
+    await setupRenderingContext(this);
+  });
+
+  hooks.afterEach(async function () {
+    await teardownRenderingContext(this);
+    this.engine = undefined;
```

Comment:
No need to reset here, but when you update to make `this.engine` a different value from `this.owner` (see other comment) you will need to destroy `this.engine`.